index.go

```diff
@@ -181,50 +192,24 @@ func (i *index) Delete(bucket *bbolt.Bucket, ref Reference, doc Document) error
 
 // addRefToBucket adds the reference to the correct key in the bucket. It handles multiple reference on the same location
 func addRefToBucket(bucket *bbolt.Bucket, key Key, ref Reference) error {
-	entryBytes := bucket.Get(key)
-	var entry Entry
-
-	if len(entryBytes) == 0 {
-		entry = EntryFrom(ref)
-	} else {
-		if err := entry.Unmarshal(entryBytes); err != nil {
-			return err
-		}
-		entry.Add(ref)
-	}
-
-	iBytes, err := entry.Marshal()
+	// first check if there's a sub-bucket
+	subBucket, err := bucket.CreateBucketIfNotExists(key)
 	if err != nil {
 		return err
 	}
-
-	return bucket.Put(key, iBytes)
+	subBucket.FillPercent = 0.9
```

Comment:
why 0.9?

Reply:
I removed those percentages, did some testing and it doesn't influence size of the DB significantly.